andela/ah-backend-odin
# #161340175 Add reset password capability

#### What does this PR do?
Allows users to reset their passwords through password-reset link and token
#### Description of Task to be completed?
The user provides his email, and a password-reset link is sent to his/her email with a token which expires after 4 hours. If the user clicks the link then he/she is granted the ability to reset his/her password.
#### How should this be manually tested?
Using Postman: run the server and visit the link `127.0.0.1:8000/api/reset_password/` and provide `"email" : "your-email"}` then POST. This will work when you are loggen in using that email. After this you should have received a reset link in your email. On Postman again: visit `127.0.0.1:8000/api/set_password/complete/` and provide the json data `"email" : "your-email", "password" : "your new password"`. If password passes all the validations then you will be able to reset your password.
#### What are the relevant pivotal tracker stories?
[#161340175](https://www.pivotaltracker.com/story/show/161340175)
